url29ff2e53f3_nicolas3470_Settlers_of_Catan_AI_tgz-pJ8-robotGameJ8/src/soc/client/SOCBoardPanel.java

```diff
@@ -0,0 +1,5144 @@
+        if (is6player)
+        {
+            secX = ((x + 13 - HEXX_OFF_6PL) / 27);
+            secY = ((y + 7 - HEXY_OFF_6PL_FIND) / 15);
+        } else {
+            secX = ((x + 13) / 27);
+            secY = ((y + 7) / 15);
+        }
+        int sector = secX + (secY * 15);
+
+        // System.out.println("SECTOR = "+sector+" | NODE = "+nodeMap[sector]);
+        if ((sector >= 0) && (sector < nodeMap.length))
+            return nodeMap[sector];
+        else
+            return 0;
+    }
+
+    /**
+     * given a pixel on the board, find the hex that contains it
+     *
+     * @param x  x coordinate, in unscaled board, not actual pixels;
+     *           use {@link #scaleFromActualX(int)} to convert
+     * @param y  y coordinate, in unscaled board, not actual pixels
+     * @return the coordinates of the hex, or 0 if none
+     */
+    private final int findHex(int x, int y)
+    {
+        // find which grid section the pointer is in 
+        // ( 46 is the y-distance between the centers of two hexes )
+        //int sector = (x / 18) + ((y / 10) * 15);
+        int secX, secY;
+        if (is6player)
+        {
+            secX = (x - HEXX_OFF_6PL) / 27;
+            secY = (y - HEXY_OFF_6PL_FIND) / 15;
+        } else {
+            secX = x / 27;
+            secY = y / 15;
+        }
+        int sector = secX + (secY * 15);
+
+        // System.out.println("SECTOR = "+sector+" | HEX = "+hexMap[sector]);
+        if ((sector >= 0) && (sector < hexMap.length))
+            return hexMap[sector];
+        else
+            return 0;
+    }
+
+    /**
+     * set the interaction mode
+     *
+     * @param m  mode
+     * 
+     * @see #updateMode()
+     */
+    public void setMode(int m)
+    {
+        mode = m;
+        updateHoverTipToMode();
+    }
+
+    /**
+     * get the interaction mode
+     *
+     * @return the mode
+     */
+    public int getMode()
+    {
+        return mode;
+    }
+
+    /**
+     * load the images for the board
+     * we need to know if this board is in an applet
+     * or an application
+     */
+    private static synchronized void loadImages(Component c, boolean wantsRotated)
+    {
+        if ((hexes != null) && ((rotatHexes != null) || ! wantsRotated))
+            return;
+
+        Toolkit tk = c.getToolkit();
+        Class clazz = c.getClass();
+
+        if (hexes == null)
+        {
+            MediaTracker tracker = new MediaTracker(c);
+        
+            hexes = new Image[13];
+            ports = new Image[7];
+            dice = new Image[14];
+
+            loadHexesPortsImages(hexes, ports, IMAGEDIR, tracker, tk, clazz);
+
+            for (int i = 2; i < 13; i++)
+            {
+                dice[i] = tk.getImage(clazz.getResource(IMAGEDIR + "/dice" + i + ".gif"));
+                tracker.addImage(dice[i], 0);
+            }
+
+            try
+            {
+                tracker.waitForID(0);
+            }
+            catch (InterruptedException e) {}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)